packages/xrpl/src/models/transactions/oracleSet.ts

```diff
@@ -142,14 +143,25 @@ export function validateOracleSet(tx: Record<string, unknown>): void {
         )
       }
 
+      const _MINIMUM_ASSET_PRICE_LENGTH = 1
+      const _MAXIMUM_ASSET_PRICE_LENGTH = 16
+
+      /* eslint-disable @typescript-eslint/no-unsafe-member-access -- we need to validate priceData.PriceData.AssetPrice value */
       if (
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- we are validating the type
         'AssetPrice' in priceData.PriceData &&
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- we are validating the type
-        !isNumber(priceData.PriceData.AssetPrice)
+        !(
+          isNumber(priceData.PriceData.AssetPrice) ||
+          (isString(priceData.PriceData.AssetPrice) &&
+            isHex(priceData.PriceData.AssetPrice) &&
+            priceData.PriceData.AssetPrice.length <=
+              _MAXIMUM_ASSET_PRICE_LENGTH &&
+            priceData.PriceData.AssetPrice.length >=
+              _MINIMUM_ASSET_PRICE_LENGTH)
+        )
       ) {
         throw new ValidationError('OracleSet: invalid field AssetPrice')
```

Comment:
This conditional looks messy. It would be better to do 3 separate conditionals with their own error messages:
1. isNumber
2. isString & isHex
3. in range

Reply:
how does this look? [e1ae16f](https://github.com/XRPLF/xrpl.js/pull/2913/commits/e1ae16fa566f9d7dc99a3d46cb94a9a695705bf4)

Reply:
Looks better but the asset price length condition doesn't have to be nested.

Reply:
Oh nevermind, it also has to be nested. The new change looks good now.